source/normalize-arguments.ts

```diff
@@ -1,536 +0,0 @@
-	}
-
-	// Better memory management, so we don't have to generate a new object every time
-	if (options.cache) {
-		(options as NormalizedOptions).cacheableRequest = new CacheableRequest(
-			// @ts-ignore Cannot properly type a function with multiple definitions yet
-			(requestOptions, handler) => requestOptions[requestSymbol](requestOptions, handler),
-			options.cache
-		);
-	}
-
-	// `options.cookieJar`
-	if (is.object(options.cookieJar)) {
-		let {setCookie, getCookieString} = options.cookieJar;
-
-		// Horrible `tough-cookie` check
-		if (setCookie.length === 4 && getCookieString.length === 0) {
-			if (!Reflect.has(setCookie, promisify.custom)) {
-				// @ts-ignore TS is dumb - it says `setCookie` is `never`.
-				setCookie = promisify(setCookie.bind(options.cookieJar));
-				getCookieString = promisify(getCookieString.bind(options.cookieJar));
-			}
-		} else if (setCookie.length !== 2) {
-			throw new TypeError('`options.cookieJar.setCookie` needs to be an async function with 2 arguments');
-		} else if (getCookieString.length !== 1) {
-			throw new TypeError('`options.cookieJar.getCookieString` needs to be an async function with 1 argument');
-		}
-
-		options.cookieJar = {setCookie, getCookieString};
-	}
-
-	// `options.encoding`
-	if (is.null_(options.encoding)) {
-		throw new TypeError('To get a Buffer, set `options.responseType` to `buffer` instead');
-	}
-
-	// `options.maxRedirects`
-	if (!Reflect.has(options, 'maxRedirects') && !(defaults && Reflect.has(defaults, 'maxRedirects'))) {
-		options.maxRedirects = 0;
-	}
-
-	// Merge defaults
-	if (defaults) {
-		options = merge({}, defaults, options);
-	}
-
-	// `options._pagination`
-	if (is.object(options._pagination)) {
-		const {_pagination: pagination} = options;
-
-		if (!is.function_(pagination.transform)) {
-			throw new TypeError('`options._pagination.transform` must be implemented');
-		}
-
-		if (!is.function_(pagination.shouldContinue)) {
-			throw new TypeError('`options._pagination.shouldContinue` must be implemented');
-		}
-
-		if (!is.function_(pagination.filter)) {
-			throw new TypeError('`options._pagination.filter` must be implemented');
-		}
-
-		if (!is.function_(pagination.paginate)) {
-			throw new TypeError('`options._pagination.paginate` must be implemented');
-		}
-	}
-
-	// Other values
-	options.decompress = Boolean(options.decompress);
-	options.isStream = Boolean(options.isStream);
-	options.throwHttpErrors = Boolean(options.throwHttpErrors);
-	options.ignoreInvalidCookies = Boolean(options.ignoreInvalidCookies);
-	options.cache = options.cache ?? false;
-	options.responseType = options.responseType ?? 'text';
-	options.resolveBodyOnly = Boolean(options.resolveBodyOnly);
-	options.followRedirect = Boolean(options.followRedirect);
-	options.dnsCache = options.dnsCache ?? false;
-	options.useElectronNet = Boolean(options.useElectronNet);
-	options.methodRewriting = Boolean(options.methodRewriting);
-	options.allowGetBody = Boolean(options.allowGetBody);
```

Comment:
This has been dropped to get a 1.5k req/s increase. It will be inherited from defaults so it's unnecessary either because the defaults define these booleans.